File: pkg/dto/jenkins.go
package dto

// TriggerBuildRequest 触发构建请求
type TriggerBuildRequest struct {
	JobName            string `json:"job_name" binding:"required"`
	GitlabSourceBranch string `json:"gitlab_source_branch"`
	ChangeType         string `json:"change_type"`
	Branch             string `json:"branch"`
	DeployType         string `json:"deploy_type"`
	ImageVersion       string `json:"image_version"`
}

// TriggerBuildResponse 触发构建响应
type TriggerBuildResponse struct {
	Success     bool   `json:"success"`
	Message     string `json:"message"`
	JobName     string `json:"job_name"`
	BuildNumber int64  `json:"build_number"`
}

// JobInfoResponse Job信息响应
type JobInfoResponse struct {
	Name        string `json:"name"`
	DisplayName string `json:"display_name"`
	URL         string `json:"url"`
	Buildable   bool   `json:"buildable"`
}

// BuildStatusResponse 构建状态响应
type BuildStatusResponse struct {
	JobName     string `json:"job_name"`
	BuildNumber int64  `json:"build_number"`
	Status      string `json:"status"`
	Timestamp   int64  `json:"timestamp"`
}

// BuildConsoleResponse 构建日志响应
type BuildConsoleResponse struct {
	JobName     string `json:"job_name"`
	BuildNumber int64  `json:"build_number"`
	Log         string `json:"log"`
}

// CreateJenkinsInstanceRequest 创建Jenkins实例请求
type CreateJenkinsInstanceRequest struct {
	Name        string `json:"name" binding:"required,min=1,max=100"`
	URL         string `json:"url" binding:"required,url,min=1,max=500"`
	Username    string `json:"username" binding:"max=100"`
	APIToken    string `json:"api_token" binding:"max=500"`
	Description string `json:"description"`
	Status      string `json:"status" binding:"required,oneof=active inactive"`
	IsDefault   bool   `json:"is_default"`
}

// UpdateJenkinsInstanceRequest 更新Jenkins实例请求
type UpdateJenkinsInstanceRequest struct {
	Name        string `json:"name" binding:"required,min=1,max=100"`
	URL         string `json:"url" binding:"required,url,min=1,max=500"`
	Username    string `json:"username" binding:"max=100"`
	APIToken    string `json:"api_token" binding:"max=500"`
	Description string `json:"description"`
	Status      string `json:"status" binding:"required,oneof=active inactive"`
	IsDefault   bool   `json:"is_default"`
}

// JenkinsInstanceResponse Jenkins实例响应
type JenkinsInstanceResponse struct {
	ID          uint   `json:"id"`
	Name        string `json:"name"`
	URL         string `json:"url"`
	Username    string `json:"username"`
	Description string `json:"description"`
	Status      string `json:"status"`
	IsDefault   bool   `json:"is_default"`
	CreatedAt   string `json:"created_at"`
	UpdatedAt   string `json:"updated_at"`
}

// JenkinsInstanceListRequest Jenkins实例列表请求
type JenkinsInstanceListRequest struct {
	Page     int    `form:"page,default=1"`
	PageSize int    `form:"page_size,default=20"`
	Keyword  string `form:"keyword"`
	Status   string `form:"status"`
}

// JenkinsInstanceListResponse Jenkins实例列表响应
type JenkinsInstanceListResponse struct {
	Total      int64                     `json:"total"`
	Page       int                       `json:"page"`
	PageSize   int                       `json:"page_size"`
	TotalPages int                       `json:"total_pages"`
	Items      []JenkinsInstanceResponse `json:"items"`
}

// FeishuAppSimple 飞书应用简单信息
type FeishuAppSimple struct {
	ID      uint   `json:"id"`
	Name    string `json:"name"`
	AppID   string `json:"app_id"`
	Project string `json:"project"`
}

// BindFeishuAppsRequest 绑定飞书应用请求
type BindFeishuAppsRequest struct {
	AppIDs []uint `json:"app_ids"`
}

// JenkinsJob Jenkins Job信息
type JenkinsJob struct {
	Name            string `json:"name"`
	URL             string `json:"url"`
	Color           string `json:"color"`
	Class           string `json:"class"`
	LastBuildNumber int64  `json:"last_build_number"`
	LastBuildResult string `json:"last_build_result"`
	LastBuildTime   string `json:"last_build_time"`
}

// JenkinsBuild Jenkins 构建信息
type JenkinsBuild struct {
	Number    int64  `json:"number"`
	Result    string `json:"result"`
	Building  bool   `json:"building"`
	Timestamp string `json:"timestamp"`
	Duration  int64  `json:"duration"`
	URL       string `json:"url"`
}

File: web/src/services/jenkins.ts
import request from './api'
import type { ApiResponse, JenkinsInstance, PaginatedResponse } from '../types'

export interface JenkinsInstanceRequest {
  page?: number
  page_size?: number
  keyword?: string
  status?: string
}

export interface CreateJenkinsInstanceRequest {
  name: string
  url: string
  username?: string
  api_token?: string
  description?: string
  status: string
  is_default?: boolean
}

export interface UpdateJenkinsInstanceRequest extends CreateJenkinsInstanceRequest {}

export const jenkinsInstanceApi = {
  list: (page = 1, pageSize = 100): Promise<ApiResponse<{ items: JenkinsInstance[]; total: number }>> => {
    return request.get('/jenkins-instances', { params: { page, page_size: pageSize } })
  },

  getInstances: (params: JenkinsInstanceRequest = {}): Promise<ApiResponse<PaginatedResponse<JenkinsInstance>>> => {
    return request.get('/jenkins-instances', { params })
  },

  getInstance: (id: number): Promise<ApiResponse<JenkinsInstance>> => {
    return request.get(`/jenkins-instances/${id}`)
  },

  getDefaultInstance: (): Promise<ApiResponse<JenkinsInstance>> => {
    return request.get('/jenkins-instances/default')
  },

  createInstance: (data: CreateJenkinsInstanceRequest): Promise<ApiResponse<JenkinsInstance>> => {
    return request.post('/jenkins-instances', data)
  },

  updateInstance: (id: number, data: UpdateJenkinsInstanceRequest): Promise<ApiResponse<JenkinsInstance>> => {
    return request.put(`/jenkins-instances/${id}`, data)
  },

  setDefaultInstance: (id: number): Promise<ApiResponse> => {
    return request.put(`/jenkins-instances/${id}/default`)
  },

  deleteInstance: (id: number): Promise<ApiResponse> => {
    return request.delete(`/jenkins-instances/${id}`)
  },

  testConnection: (id: number): Promise<ApiResponse<ConnectionTestResult>> => {
    return request.post(`/jenkins-instances/${id}/test-connection`)
  },

  getFeishuApps: (id: number): Promise<ApiResponse<FeishuAppSimple[]>> => {
    return request.get(`/jenkins-instances/${id}/feishu-apps`)
  },

  bindFeishuApps: (id: number, appIds: number[]): Promise<ApiResponse> => {
    return request.put(`/jenkins-instances/${id}/feishu-apps`, { app_ids: appIds })
  }
}

export interface ConnectionTestResult {
  connected: boolean
  version?: string
  server_version?: string
  node_count?: number
  response_time_ms: number
  error?: string
}

export interface FeishuAppSimple {
  id: number
  name: string
  app_id: string
  project: string
}

export interface JenkinsJob {
  name: string
  url: string
  color: string
  class: string
  last_build_number: number
  last_build_result: string
  last_build_time: string
}

export interface JenkinsBuild {
  number: number
  result: string
  building: boolean
  timestamp: string
  duration: number
  url: string
}

export const jenkinsJobApi = {
  getJobs: (instanceId: number): Promise<ApiResponse<JenkinsJob[]>> => {
    return request.get(`/jenkins-instances/${instanceId}/jobs`)
  },

  getJobBuilds: (instanceId: number, jobName: string, limit = 20): Promise<ApiResponse<JenkinsBuild[]>> => {
    return request.get(`/jenkins-instances/${instanceId}/jobs/${encodeURIComponent(jobName)}/builds`, { params: { limit } })
  }
}

// Jenkins 构建相关
export const jenkinsApi = {
  triggerBuild: (data: { job_name: string; gitlab_source_branch?: string; change_type?: string }): Promise<ApiResponse> => {
    return request.post('/jenkins/build', data)
  },

  getBuildStatus: (jobName: string, buildNumber: number): Promise<ApiResponse> => {
    return request.get(`/jenkins/build/${jobName}/${buildNumber}/status`)
  },

  getBuildConsole: (jobName: string, buildNumber: number): Promise<ApiResponse> => {
    return request.get(`/jenkins/build/${jobName}/${buildNumber}/console`)
  },

  getJobInfo: (jobName: string): Promise<ApiResponse> => {
    return request.get(`/jenkins/job/${jobName}`)
  }
}
